storage/memory/log_storage.go

```diff
@@ -342,6 +342,22 @@ func (t *logTreeTX) GetActiveLogIDs(ctx context.Context) ([]int64, error) {
 	return t.getActiveLogIDs(ctx)
 }
 
+func (t *readOnlyLogTX) GetUnsequencedCounts(ctx context.Context) (map[int64]int64, error) {
+	t.ms.mu.RLock()
+	defer t.ms.mu.RUnlock()
+
+	ret := make(map[int64]int64)
+	for id, tree := range t.ms.trees {
+		tree.mu.RLock()
```

Comment:
Ah - thanks - I forgot about those methods - changed.

`ms.mu` really only protects the map of trees, I guess in theory it _could_ protect all the trees too, but I think there are probably places where all you have is your local `*tree`, so you have no way of getting to `ms.mu`.